lib/qmp.ml

```diff
@@ -215,10 +228,18 @@ let message_of_string str =
       let fdset_id = json |> U.member "fdset-id" |> U.to_int in
       {fd; fdset_id}
     in
+    let qom json =
+      let mem_of k x = x |> U.member k |> U.to_string in
+      json |> U.convert_each (fun x -> {name = (mem_of "name" x);ty = (mem_of "type" x)})
+    in
     let result =
       let return = json |> U.member "return" in
       return |> function
-      | `List  _ -> return |> name_list |> fun x -> Name_list x
+      | `List  _ -> (return |> U.convert_each (fun x -> x |> U.keys) |> List.flatten |> function
```

Comment:
It's unnecessarily long. The following should work as well

```ocaml
| `List  _ -> (return |> U.convert_each U.keys |> List.flatten |> function`
```

Reply:
updated.